File: 45.jump-game-ii/solution.kt
class Solution {
    fun jump(nums: IntArray): Int {
        val k = nums.size - 1
        val dp = IntArray(nums.size)
        for (i in 1..k) {
            dp[i] = Int.MAX_VALUE
            for (j in 0 until i) {
                if (j + nums[j] >= i) {
                    dp[i] = Math.min(dp[i], dp[j] + 1)
                }
            }
        }
        return dp[k]
    }
}

File: 45.jump-game-ii/solution.go
func jump(nums []int) int {
    far := 0
    end := 0
    count := 0
    for i := 0; i < len(nums) - 1; i++ {
        if i + nums[i] > far {
            far = i + nums[i]
        }
        if i == end {
            count++
            end = far
        }
    }
    return count
}